lib/grpcServer/handlers/tx-filter-stream/subscribeToTransactionsWithProofsHandlerFactory.js

```diff
@@ -121,55 +121,52 @@ function subscribeToTransactionsWithProofsHandlerFactory(
 
     const isNewTransactionsRequested = count === 0;
 
+    const acknowledgingCall = new AcknowledgingWritable(call);
+
+    const mediator = new ProcessMediator();
+
+    if (isNewTransactionsRequested) {
+      subscribeToNewTransactions(
+        mediator,
+        filter,
+        testTransactionAgainstFilter,
+        bloomFilterEmitterCollection,
+      );
+    }
+
+    // If block height is specified instead of block hash, we obtain block hash by block height
     let blockHash = fromBlockHash;
 
     if (blockHash === '') {
       if (fromBlockHeight === 0) {
         throw new InvalidArgumentGrpcError('minimum value for `fromBlockHeight` is 1');
       }
 
-      const bestHeight = await coreAPI.getBestBlockHeight();
-
-      if (fromBlockHeight > bestHeight) {
-        throw new InvalidArgumentGrpcError('fromBlockHeight is bigger than block count');
-      }
-
+      // we don't need to check bestBlockHeight because getBlockHash throws
+      // an error in case of wrong height
       blockHash = await coreAPI.getBlockHash(fromBlockHeight);
```

Comment:
We need to catch error and respond with `InvalidArgumentGrpcError` otherwise it will be an internal error.